pkgs/applications/networking/remote/x2goserver/xsession-nixos-support.patch

```diff
@@ -0,0 +1,22 @@
+diff --git a/x2goserver-xsession/etc/Xsession b/x2goserver-xsession/etc/Xsession
+index ebee0c5f..866f763e 100755
+--- a/x2goserver-xsession/etc/Xsession
++++ b/x2goserver-xsession/etc/Xsession
+@@ -145,7 +145,7 @@ if [ -f /etc/debian_version ] || [ -f /etc/devuan_version ]; then
+ 		done
+ 		set -e
+ 	fi
+-elif [ -f /etc/redhat-release ] || [ -f /etc/gentoo-release ] || [ -f /etc/SUSE-brand ] || [ -f /etc/SuSE-release ] || [ -f /etc/os-rt-release ]; then
++elif [ -f /etc/redhat-release ] || [ -f /etc/gentoo-release ] || [ -f /etc/SUSE-brand ] || [ -f /etc/SuSE-release ] || [ -f /etc/os-rt-release ] || [ -d /etc/nixos ]; then
+ 
+ 	# define a fallback... (should never be needed). The XSESSION_EXEC var gets set in
+ 	# X2Go's x2goruncommand script and can be used with obsolete switchdesk or with
+@@ -232,7 +232,7 @@ elif [ -f /etc/redhat-release ] || [ -f /etc/gentoo-release ] || [ -f /etc/SUSE-
+ 	#	exec /bin/bash -c "exec -l \"$SHELL\" -c \"$SWITCHDESKPATH/Xclients.${XSESSION_EXEC}\""
+ 	#fi
+ 
+-	exec $CK_XINIT_SESSION $SSH_AGENT /bin/bash -c "exec -l \"$SHELL\" -c \"$STARTUP\""
++	exec $CK_XINIT_SESSION $SSH_AGENT /run/current-system/sw/bin/bash -c "exec -l \"$SHELL\" -c \"$STARTUP\""
```

Comment:
Wouldn't this patch make x2goserver fail on other distros? Nixpkgs packages need to be distro-agnostic

Reply:
I'll replace this with ${bash}/bin/bash and take out the patch as well.